llmproxy/llmproxy.py

```diff
@@ -185,3 +184,52 @@ def _cost_route(self, prompt: str):
             )
 
         return completion_res
+    
+    def _category_route(self, prompt:str):
+        min_heap = MinHeap()
+        classification_model = CategoryModel(prompt=prompt)
+        best_fit_category = classification_model.categorize_text()
+        for model_name, instance, in self.user_models.items():
+            logger.info(msg="========Start Category Routing===========\n")
+            category_rank = instance.get_category_rank(best_fit_category)
+            item = {"name": model_name, "rank": category_rank, "instance": instance}
+            min_heap.push(category_rank, item)
+            logger.info(msg="SORTING MODELS...\n")
+            logger.info(msg="========End Category Routing=============\n")
+
+        completion_res = None
+        while not completion_res:
+            # Iterate through heap until there are no more options
+            min_val_instance = min_heap.pop_min()
```

Comment:
Does a lower number for category mean better or more of a match? 
- If yes, then the logic seems reversed
- If no, then you shouldn't be using a min heap

Reply:
Lower the number means better of a match (e.g. 1st, 2nd, 3rd, etc.). The algo was made that way to reuse code/logic, like the minheap, to make understanding the codebase easier. If you still think it should be changed I could make a max heap instead, and have the ranking be 5 is best and 1 is worst.

Reply:
But what happens when we add more models, wouldn't there be an overlap in values?